src/main.rs

```diff
@@ -445,6 +467,84 @@ fn analog_write(spi_drv: &mut SpiDrv, uart: &mut EnabledUart, pin: u8, value: u8
     spi_drv.esp_deselect();
 }
 
+fn wifi_set_passphrase(spi_drv: &mut SpiDrv, uart: &mut EnabledUart, mut ssid: String<STR_LEN>, mut passphrase: String<STR_LEN>) -> bool {
+    spi_drv.wait_for_esp_select();
+
+    // Send Command
+    spi_drv.send_cmd(uart, SET_PASSPHRASE, 2).ok().unwrap();
+
+    // FIXME: for the real crate, don't use unsafe
+    let ssid_bytes: &mut [u8] = unsafe { ssid.as_bytes_mut() };
+    writeln!(uart, "ssid: {:?}\r", ssid_bytes).ok().unwrap();
+    spi_drv.send_param(uart, ssid_bytes, false).ok().unwrap();
+
+    // FIXME: for the real crate, don't use unsafe
+    let passphrase_bytes: &mut [u8] = unsafe { passphrase.as_bytes_mut() };
+    writeln!(uart, "passphrase: {:?}\r", passphrase_bytes).ok().unwrap();
+    spi_drv.send_param(uart, passphrase_bytes, true).ok().unwrap();
+
+    let command_size: u8 = 6 + ssid.len() as u8 + passphrase.len() as u8;
+    spi_drv.pad_to_multiple_of_4(uart, command_size);
+ 
+    spi_drv.esp_deselect();
+    spi_drv.wait_for_esp_select();
+
+    // Wait for reply
+    let data: u8 = 0;
+    let wait_response = spi_drv.wait_response_cmd(uart, SET_PASSPHRASE, 1);
+    match wait_response {
+        Ok(params) => {
+            write!(uart, "\twifi_set_passphrase_response: ").ok().unwrap();
+            for byte in params {
+                let c = byte as char;
+                write!(uart, "{:?}", c).ok().unwrap();
+            }
+            writeln!(uart, "\r\n").ok().unwrap();
+        }
+        Err(e) => {
+            writeln!(uart, "\twifi_set_passphrase_response Err: {:?}\r", e).ok().unwrap();
+            spi_drv.esp_deselect();
+            return false;
+        }
+    }
+
+    spi_drv.esp_deselect();
+
+    true
+}
+
+fn get_connection_status(spi_drv: &mut SpiDrv, uart: &mut EnabledUart) -> bool {
+    spi_drv.wait_for_esp_select(); 
+   
+    // Send Command
+    spi_drv.send_cmd(uart, GET_CONN_STATUS, 0).ok().unwrap();
+
+    spi_drv.esp_deselect();
+    spi_drv.wait_for_esp_select();
+
+    // Wait for reply
+    let data: u8 = 0;
+    let wait_response = spi_drv.wait_response_cmd(uart, GET_CONN_STATUS, 1);
+    match wait_response {
+        Ok(params) => {
+            write!(uart, "\tget_connection_status_response: ").ok().unwrap();
+            for byte in params {
+                let c = byte as char;
+                write!(uart, "{:?}", c).ok().unwrap();
+            }
+            writeln!(uart, "\r\n").ok().unwrap();
+        }
+        Err(e) => {
+            writeln!(uart, "\tget_connection_status_response Err: {:?}\r", e).ok().unwrap();
```

Comment:
```suggestion
            writeln!(uart, "\tget_connection_status_response Err: {:?}\r", e).ok().unwrap();
            return false;
```

Reply:
Resolved in [following commit](https://github.com/Jim-Hodapp-Coaching/esp32-pico-wifi/pull/4/commits/c10bb540c941bdb44dbff93ce28af3736f1ca7b1)